### PR TITLE
Adding rule NL.BR-KVK-4.17 and related test.

### PR DESCRIPTION
**Pull request by allenbross-wf**
#### Reason for change
BR-KVK-4.17: The current and previous financial reporting period MUST be less than 2 years.

#### Description of change

Added a rule to check the previous and current reporting periods. 

#### Steps to Test
CI

**review**:
@Arelle/arelle

### CODE REVIEW

**Comment by aviary2-wf** [comment]
## Security Insights
No security relevant content was detected by automated scans.

## Action Items
* Review PR for [security impact](https://wiki.atl.workiva.net/display/SECURITY/Development+Security+Review+Guidelines); comment "security review required" if needed or unsure
* Verify [`aviary.yaml`](https://wiki.atl.workiva.net/pages/viewpage.action?pageId=42762752#SecurityInsightFramework&Tooling-ConfiguringAviary) coverage of security relevant code
---
_Questions or Comments? Reach out on Slack: #support-infosec._

File: tests/resources/conformance_suites/nl_nt17/br_kvk/4-17-testcase.xml [hunk 1]
```diff
@@ -0,0 +1,22 @@
...
+  xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance"
+  xmlns="http://xbrl.org/2005/conformance"
+  name="NL.BR-KVK-4.17"
+  description="All financial reporting periods must be shorter than 2 years"
+  outpath=''
+  owner="support@arelle.org"
+  xsi:schemaLocation="http://xbrl.org/2005/conformance https://www.xbrl.org/2005/conformance.xsd">
+  <variation id="period-duration" name="Invalid Financial Reporting Period Duration">
+    <description>
+      The instance has and invalid current and previous financial reporting periods.
```
Comment (aaroncameron-wk) [inline tests/resources/conformance_suites/nl_nt17/br_kvk/4-17-testcase.xml:13]:
```suggestion
      The instance has invalid current and previous financial reporting periods.
```

File: tests/resources/conformance_suites/nl_nt17/br_kvk/4-17-invalid-period.xbrl [hunk 2]
```diff
@@ -0,0 +1,55 @@
...
+  xmlns:link="http://www.xbrl.org/2003/linkbase"
+  xmlns:nl-cd="http://www.nltaxonomie.nl/nt16/sbr/20210301/dictionary/nl-common-data"
+  xmlns:rj-dim="http://www.nltaxonomie.nl/nt16/rj/20211208/dictionary/rj-axes"
+  xmlns:rj-dm="http://www.nltaxonomie.nl/nt16/rj/20211208/dictionary/rj-domains"
+  xmlns:rj-i="http://www.nltaxonomie.nl/nt16/rj/20211208/dictionary/rj-data"
+  xmlns:xbrldi="http://xbrl.org/2006/xbrldi"
+  xmlns:xbrli="http://www.xbrl.org/2003/instance"
+  xmlns:xlink="http://www.w3.org/1999/xlink">
+    <link:schemaRef
+      xlink:href="http://www.nltaxonomie.nl/nt16/kvk/20211208/entrypoints/kvk-rpt-jaarverantwoording-2021-nlgaap-klein-publicatiestukken.xsd"
```
Comment (aaroncameron-wk) [inline tests/resources/conformance_suites/nl_nt17/br_kvk/4-17-invalid-period.xbrl:20]: This is a NL17 conformance suite test case but you're importing NL16 here and in the imports above. You can copy and paste the corresponding lines from most any other existing NL17 test case.

File: tests/resources/conformance_suites/nl_nt16/br_kvk/4-17-testcase.xml [hunk 3]
```diff
@@ -0,0 +1,22 @@
...
+  xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance"
+  xmlns="http://xbrl.org/2005/conformance"
+  name="NL.BR-KVK-4.17"
+  description="All financial reporting periods must be shorter than 2 years"
+  outpath=''
+  owner="support@arelle.org"
+  xsi:schemaLocation="http://xbrl.org/2005/conformance https://www.xbrl.org/2005/conformance.xsd">
+  <variation id="period-duration" name="Invalid Financial Reporting Period Duration">
+    <description>
+      The instance has and invalid current and previous financial reporting periods.
```
Comment (aaroncameron-wk) [inline tests/resources/conformance_suites/nl_nt16/br_kvk/4-17-testcase.xml:13]:
```suggestion
      The instance has invalid current and previous financial reporting periods.
```

File: arelle/plugin/validate/NL/rules/br_kvk.py [hunk 4]
```diff
@@ -282,3 +283,41 @@ def rule_br_kvk_4_16(
                 resubmissionConceptQname=resubmissionConceptQname,
                 conceptQname=conceptQname,
             )
+
```
Comment (aaroncameron-wk) [inline arelle/plugin/validate/NL/rules/br_kvk.py:286]:
```suggestion


```
Nit: my IDE suggests two blank lines in this context. [Underlying flake8 rule](https://www.flake8rules.com/rules/E302.html)

Comment (derekgengenbacher-wf) [reply arelle/plugin/validate/NL/rules/br_kvk.py:286]: I hate flake8 more than anything in the world, well maybe not as much as the Griz, but thats neither here or there. I do however, agree with Aaron's sentiment on this. 2 lines please.

**Review by derekgengenbacher-wf — DISMISSED** [review]

**Review by aaroncameron-wk — APPROVED** [review]

**Review by derekgengenbacher-wf — APPROVED** [review]

**Review by aaroncameron-wk — APPROVED** [review]

**Review by austinmatherne-wk — APPROVED** [review]

File: arelle/plugin/validate/NL/rules/br_kvk.py [hunk 5]
```diff
@@ -4,6 +4,7 @@
 from __future__ import annotations
 
 from datetime import date, timedelta
+from dateutil import relativedelta
```
Comment (austinmatherne-wk) [inline arelle/plugin/validate/NL/rules/br_kvk.py:7]: This looks like it needs to be explicitly required in pyproject.toml and requirements.txt

Comment (austinmatherne-wk) [inline arelle/plugin/validate/NL/rules/br_kvk.py:7]: This looks like it needs to be explicitly required in pyproject.toml and requirements.txt

File: pyproject.toml [hunk 6]
```diff
@@ -29,6 +29,7 @@ classifiers = [
 requires-python = ">=3.8"
 dependencies = [
     'certifi',
+    'python-dateutil==2.8.*',
```
Comment (austinmatherne-wk) [inline pyproject.toml:32]: Any major matching version should be okay
```suggestion
    'python-dateutil==2.*',
```

**Review by aaroncameron-wk — APPROVED** [review]

**Review by derekgengenbacher-wf — APPROVED** [review]

**Comment by derekgengenbacher-wf** [comment]
QA +10 Testing locally succeeded.

**Comment by derekgengenbacher-wf** [comment]
QA +10 Testing locally succeeded.

**Pull request merged**